homeassistant/components/irm_kmi/config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Config flow to set up IRM KMI integration via the UI."""
+
+import asyncio
+import logging
+
+from irm_kmi_api import IrmKmiApiClient
+import voluptuous as vol
+
+from homeassistant.components.zone import DOMAIN as ZONE_DOMAIN
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult, OptionsFlow
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_ZONE
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    EntitySelector,
+    EntitySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CONF_LANGUAGE_OVERRIDE,
+    CONF_LANGUAGE_OVERRIDE_OPTIONS,
+    CONFIG_FLOW_VERSION,
+    DOMAIN,
+    OUT_OF_BENELUX,
+    USER_AGENT,
+)
+from .types import IrmKmiConfigEntry
+from .utils import get_config_value
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IrmKmiConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Configuration flow for the IRM KMI integration."""
+
+    VERSION = CONFIG_FLOW_VERSION
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: IrmKmiConfigEntry) -> OptionsFlow:
+        """Create the options flow."""
+        return IrmKmiOptionFlow(config_entry)
+
+    async def async_step_user(self, user_input: dict | None = None) -> ConfigFlowResult:
+        """Define the user step of the configuration flow."""
+        errors = {}
+
+        if user_input:
+            _LOGGER.debug("Provided config user is: %s", user_input)
+
+            if (zone := self.hass.states.get(user_input[CONF_ZONE])) is None:
+                errors[CONF_ZONE] = "zone_not_exist"
+
+            # Check if zone is in Benelux.
+            if not errors:
+                assert zone is not None  # Assert is here for mypy linting.
+                api_data = {}
+                try:
+                    async with asyncio.timeout(60):
```

Comment:
ideally timeout is part of the library

Reply:
It is already done by the library but somehow I left them here as well.  Fixed in commit 8ece5680f243ce0a20e75b20d3e68cd3e82e07b7